From 642347f1b8ec9086c13fe7abd5dfef16ecbe4503 Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Tue, 23 May 2017 10:24:37 +0300
Subject: [PATCH 2/3] Add option to skip setting RPATH

OMPL installs to /usr/lib by default and some projects integrating
OMPL trigger a QA check suggesting that the set RPATH is useless.
For example Yocto gives this warning:

    QA Issue: ompl: /work/corei7-64-refkit-linux/ompl/1.1.0-r0/packages-split/ompl/usr/lib/libompl.so.1.1.0 contains probably-redundant RPATH /usr/lib [useless-rpaths]

Add the option OMPL_SKIP_RPATH that helps to skip setting RPATH
for the library. By default the option is OFF.

Upstream-Status: Backport [expected in version >1.3.1]

Commit https://github.com/ompl/ompl/commit/5deb2275e13171f8 was
cherry-picked and adjusted for version 1.2.2.

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 CMakeModules/CompilerSettings.cmake | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/CMakeModules/CompilerSettings.cmake b/CMakeModules/CompilerSettings.cmake
index cba0f87..e0b300b 100644
--- a/CMakeModules/CompilerSettings.cmake
+++ b/CMakeModules/CompilerSettings.cmake
@@ -52,11 +52,14 @@ if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
     add_definitions(-fPIC)
 endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
 
-# Set rpath http://www.paraview.org/Wiki/CMake_RPATH_handling
-set(CMAKE_SKIP_BUILD_RPATH FALSE)
-set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+option(OMPL_SKIP_RPATH "Don't set RPATH to the OMPL library" OFF)
+if(NOT OMPL_SKIP_RPATH)
+    # Set rpath http://www.paraview.org/Wiki/CMake_RPATH_handling
+    set(CMAKE_SKIP_BUILD_RPATH FALSE)
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+endif()
 
 # no prefix needed for python modules
 set(CMAKE_SHARED_MODULE_PREFIX "")
-- 
2.7.4

